tokio/src/task/local.rs

```diff
@@ -360,12 +362,16 @@ const MAX_TASKS_PER_TICK: usize = 61;
 const REMOTE_FIRST_INTERVAL: u8 = 31;
 
 /// Context guard for LocalSet
-pub struct LocalEnterGuard(Option<Rc<Context>>);
+pub struct LocalEnterGuard {
+    ctx: Option<Rc<Context>>,
+    entered: bool,
```

Comment:
can we add a comment here describing how `entered` is used and what it represents? thanks!

Reply:
I've added! Thanks!

Reply:
Sorry if this is overly nitpicky, but I think this comment could maybe be improved a bit --- it would be nice if it explained *why* we need to differentiate between `enter` and polling the `LocalSet`, and what behavior is controlled based on that. I don't think that's currelty all that clear from reading the code.

Reply:
Oops, sorry! I've corrected it!